### PR TITLE
Add a scale conversion function

### PR DESCRIPTION
**Pull request by jhkennedy**
Will need to switch from Power to db when doing the expectation maximization thresholding.

Of note: This is based on what @hjkristenson did for the ArcGIS toolbox, but doesn't look exactly like the db conversion in the notebooks:

```python
db = np.log10(power)+30
```

### CODE REVIEW

File: asf_tools/raster.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import warnings
+import numpy as np
+
+
+def convert_scale(array: np.ndarray, in_scale: str, out_scale: str) -> np.ndarray:
```
Comment (asjohnston-asf) [inline asf_tools/raster.py:5]: we're supporting python 3.8+, so we could use `in_scale: Literal['db', 'amplitude', 'power']` here

**Comment by hjkristenson** [comment]
> Will need to switch from Power to db when doing the expectation maximization thresholding.
> 
> Of note: This is based on what @hjkristenson did for the ArcGIS toolbox, but doesn't look exactly like the db conversion in the notebooks:
> 
> ```python
> db = np.log10(power)+30
> ```

This is an unexpected conversion. Usually it's 10 * log10(power).
Are you saying that the above is what's in my ArcGIS Toolbox or what's in the notebook?

File: tests/test_raster.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    c = raster.convert_scale(np.array([np.nan, -np.inf, 0., 6.02059991, 9.54242509]), 'db', 'power')
+    assert np.isnan(c[0])
+    assert np.all(np.isclose(np.array([0, 1, 4, 9]), c[1:]))
+
+    c = raster.convert_scale(np.array([-np.inf, -20., 0., 13.97940009, 20.]), 'db', 'amplitude')
+    assert np.all(np.isclose(np.array([0., 0.1, 1., 5., 10.]), c))
+
+    a = np.array([-10, -5, 0, 5, 10])
+    with pytest.raises(ValueError):
+        _ = raster.convert_scale(a, 'power', 'foo')
```
Comment (asjohnston-asf) [inline tests/test_raster.py:31]: I typically don't bother calling out the return value in `with pytest.raises` blocks.  I'm kinda on the fence about it, how strongly do you feel one way or the other?

**Comment by hjkristenson** [comment]
Ah, I see that you're using the 10*log10(power) conversion in your code. I've not seen the log10(power)+30 before; would be worth checking with the notebook authors to see if they've found that to be a better scale for what they're doing.

File: tests/test_raster.py (hunk 2 above)
Comment (jhkennedy) [reply tests/test_raster.py:31]: Done to get PyCharm to stop complaining about "Statement has no effect"

**Comment by jhkennedy** [comment]
After speaking with Franz, the `log10(power) + 30` is to switch to a logarithm scale that fits in the 0 -- 256 range as expected by the expectation maximization thresholding, so it's a psedo-db scale for that application specifically. So, for this fn, I think we're g2g

**Review by asjohnston-asf — APPROVED** [review]

**Pull request merged**